knowledge-base/custom-rendering-extension.md

```diff
@@ -0,0 +1,127 @@
+---
+title: How to Create Custom Rendering Extension
+description: Implementing Custom IRenderingExtension
+type: how-to
+page_title: Custom IRenderingExtension
+slug: custom-rendering-extension
+position: 
+tags: 
+ticketid: 1461614
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+In the forum thread [Export to XML](https://www.telerik.com/forums/export-to-xml) you can find a sample implementation of custom 
+rendering extension. Here we explain some important requirements that you need to meet in order for the rendering extension 
+to work as expected. 
+
+## Solution
+In the configuration file of the application, it is necessary to register the Custom Extension:
+```XML
+<Telerik.Reporting>
+    <Extensions>
+      <Render>
+        <Extension name="CustomExtensionName" 
+                   type="NameSpace.CustomRenderingExtension, ProjectName" 
+                   description="This Value Will Appear As Extension Name In the Export Options of the Viewers and Designers">
+        </Extension>
+      </Render>
+    </Extensions>
+  </Telerik.Reporting>
+```
+
+Here is sample code of the _NameSpace.CustomRenderingExtension_ type implementing the 
+[IRenderingExtension](../t-telerik-reporting-processing-irenderingextension) interface, adapted from the example in the above 
+forum thread.
+```C#
+namespace NameSpace
+{
+    public class CustomRenderingExtension : Telerik.Reporting.Processing.IRenderingExtension
+    {
+        static bool ShouldCountOnly(IDictionary renderingContext)
+        {
+            if (renderingContext.Contains("CountOnly"))
+            {
+                return (bool)renderingContext["CountOnly"];
+            }
+
+            return false; // Default
+        }
+
+        public bool Render(Telerik.Reporting.Processing.Report report
+            , System.Collections.Hashtable renderingContext
+            , System.Collections.Hashtable deviceInfo
+            , Telerik.Reporting.Processing.CreateStream createStreamCallback
+            , Telerik.Reporting.Processing.EvaluateHeaderFooterExpressions evalHeaderFooterCallback)
+        {
+            if (ShouldCountOnly(renderingContext))
+            { // I do not report page count since it does not make any sense.
```

Comment:
Well, if I write an extension producing pages, this comment would be out of order. Because it makes sense. Instead, as a minimum change the comment to:
Our custom rendering extension format (XML) does not have pages, so we do not need to count pages.\
...or something similar.